sync/src/types.rs

```diff
@@ -396,18 +398,20 @@ impl Peers {
         peer: PeerIndex,
         headers_sync_timeout: Option<u64>,
         protect: bool,
-        is_outbound: bool,
+        (is_outbound, is_whitelist): (bool, bool),
```

Comment:
Why tuple? flat args is better.

Reply:
Did not see the obvious benefits and disadvantages, where is better?

Reply:
I don't think there is any essential difference between the two in terms of definition.

Reply:
We can randomly put two parameters into a tuple and there not any "essential difference", and the program still works, but why we do this? 

Why don't u put other parameters into a tuple but these two? Does it imply the inner link of these two? If it is what's the hidden reason? And if it is not, why don't we put other parameters into tuple `(protect, is_outbound, is_whitelist)`?

Unreasonable change is a disadvantage.

Reply:
Why is `protect` not put together, because protect is the self-judgment of the protocol layer, and the other two are the state of `nc`. 

I am going to write it according to the @quake  approach.